OpenDreamRuntime/Objects/Types/DreamList.cs

```diff
@@ -372,6 +372,8 @@ public override DreamValue OperatorEquivalent(DreamValue b) {
             return DreamValue.False;
         if (GetLength() != secondList.GetLength())
             return DreamValue.False;
+        if(IsAssociative && (!secondList.IsAssociative || secondList.GetAssociativeValues().Count != GetAssociativeValues().Count))
```

Comment:
`list(a=null) ~= list("a")` is `TRUE` so returning early if `IsAssociative != secondList.IsAssociative` is incorrect. At least as long as `IsAssociative` doesn't take all-null associative lists into account.